libs/knowledge-store/ragstack_knowledge_store/graph_store.py

```diff
@@ -683,23 +721,24 @@ def _get_outgoing_tags(
 
         def add_sources(rows: Iterable[Any]) -> None:
             for row in rows:
-                tags.update(row.link_to_tags)
+                if row.link_to_tags:
+                    tags.update(row.link_to_tags)
 
         with self._concurrent_queries() as cq:
             for source_id in source_ids:
                 cq.execute(
-                    self._query_source_tags_by_id, (source_id,), callback=add_sources
+                    self._query_ids_and_link_to_tags_by_id, (source_id,), callback=add_sources
                 )
 
         return tags
 
     def _get_adjacent(
         self,
-        tags: set[tuple[str, str]],
+        tags: Set[Tuple[str, str]],
```

Comment:
Fixing type annotations. Upper case types (from `typing`) actually type check, while lower case types (from builtins) ignore the parameters, etc.

Also fixing the `B006` errors since those can cause problems.

Reply:
Actually, it seems like `set` and `tuple` may support type arguments as of 3.9. So I can probably back out those changes and just keep the fix for `B006`.

Reply:
Yes. [PEP585](https://peps.python.org/pep-0585/) and [PEP604](https://peps.python.org/pep-0604/) are supported (through `from __future__ import annotations` for the latter) and enforced.
The only case where we shouldn't use them and ignore the rule is with pydantic v1.
Pydantic v1 is EOL'd so we must not use it.

Reply:
Langchain requires Pydantic v1, so we must use it.